src/Symfony/Component/HttpKernel/DependencyInjection/ControllerArgumentValueResolverPass.php

```diff
@@ -40,9 +43,19 @@ public function process(ContainerBuilder $container)
             return;
         }
 
+        $resolvers = $this->findAndSortTaggedServices($this->argumentValueResolverTag, $container);
+
+        if ($container->getParameter('kernel.debug') && class_exists(Stopwatch::class)) {
+            foreach ($container->findTaggedServiceIds('controller.argument_value_resolver') as $id => $tags) {
+                $container->register($id.'.traceable', TraceableValueResolver::class)
+                    ->setDecoratedService($id)
+                    ->setArguments(array(new Reference($id.'.traceable.inner'), new Reference('debug.stopwatch')));
```

Comment:
this binds the feature to FrameworkBundle as it's the one registering `debug.stopwatch`. I would make it `null` if invalid and instantiates `StopWatch` in the TraceableValueResolver constructor if not injected

Reply:
The class is kinda useless if null is inserted though :sweat_smile: there's no way of fetching the stopwatch. The only thing that happens, is that it doesn't crash on NULL.